stdlib/email/headerregistry.pyi

```diff
@@ -141,9 +141,20 @@ if sys.version_info >= (3, 8):
         @staticmethod
         def value_parser(value: str) -> MessageID: ...
 
+class _HeaderParser(Protocol):
+    max_count: int | None
+    @staticmethod
+    def value_parser(value: str) -> Any: ...
```

Comment:
All the `value_parser` methods for all the various `Header` classes return a class that's a subclass of `email._header_value_parser.TokenList`

```suggestion
    def value_parser(value: str) -> TokenList: ...
```